packages/lib/survey/service.ts

```diff
@@ -399,13 +427,14 @@ export async function deleteSurvey(surveyId: string) {
     select: selectSurvey,
   });
 
-  revalidateTag(getSurveysCacheTag(deletedSurvey.environmentId));
-  revalidateTag(getSurveyCacheTag(surveyId));
-
   responseCache.revalidate({
     surveyId,
     environmentId: deletedSurvey.environmentId,
   });
+  surveyCache.revalidate({
+    id: deletedSurvey.id,
```

Comment:
also when a survey gets deleted we need to update all cache tags including byAttributeId and byActionClassId

Reply:
Done